app/src/main/java/com/eis/phoneringer/structure/AppManager.java

```diff
@@ -0,0 +1,100 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+import android.media.Ringtone;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.eis.phoneringer.exceptions.WrongPasswordException;
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+/**
+ * This is a singleton class used to manage a received RingCommand or to send one
+ *
+ * @author Alberto Ursino, Luca Crema, Alessandra Tonin, Marco Mariotto
+ */
+public class AppManager {
+
+    private static final int TIMEOUT_TIME = 30 * 1000; //30 seconds
+
+    /**
+     * Instance of the class that is instantiated in getInstance method
+     */
+    private static AppManager instance = null;
+
+    /**
+     * Private constructor
+     */
+    private AppManager() {
+    }
+
+    /**
+     * @return the AppManager instance
+     */
+    public static AppManager getInstance() {
+        if (instance == null)
+            instance = new AppManager();
+        return instance;
+    }
+
+    /**
+     * If the password of the RingCommand received is valid then play ringtone for fixed amount of time
+     *
+     * @param context     of the application
+     * @param ringCommand received
+     * @param ringtone    A valid ringtone to be played
+     * @throws WrongPasswordException Exception thrown when the password received is not valid
+     */
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
```

Comment:
1. Why is this required? 
2. What happens if someone has KitKat? Not saying that you should handle that case, but probably if this method doesn't work and there is no alternative method for lower APIs, the whole app doesn't make sense either. It'd probably be good to say that minimum API required is 5.0 on `build.gradle` and remove this annotation

Reply:
Method `playRingtone` of the class `RingtoneHandler` requires that API

Reply:
I'm fixing that problem with a modification in the file `build.grandle`

Reply:
API 5.0 doesn't exist. Minimin level is 7...

Reply:
Man, 5.0/Lollipop is build no. 21 https://source.android.com/setup/start/build-numbers

Reply:
lvl 21 maybe

Reply:
Sorry I was thinking about the levels not the Api version